webapp/lib/controller.dart

```diff
@@ -256,9 +257,18 @@ void listenForDirectoryMetrics() {
   );
 }
 
-bool isProjectStale(model.NeedsReplyData projectData) {
+bool isDataStale(Object projectData) {
+  var data;
+  if (projectData is model.NeedsReplyData) {
+    data = projectData as model.NeedsReplyData;
+  } else if (projectData is model.SystemMetricsData) {
+    data = projectData as model.SystemMetricsData;
+  } else {
+    throw new model.ModelDoesNotExistException('Model does not exist');
```

Comment:
let's make this error message more helpful - I think something like is a better description of what's going on, and gives the person reading the error something to get started debugging with:
```suggestion
    throw new model.DataModelNotSupported('Data object of type "${projectData.runtimeType}" not supported for staleness monitoring');
```